index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
+  //.
+  //. Takes a unary predicate, a unary function, and a value of any type, and
+  //. returns the result of applying the function to the value if the value
+  //. satisfies the predicate; the value otherwise.
+  //.
+  //. ```javascript
+  //. > S.when(x => x < 0, Math.abs, -1)
+  //. 1
+  //.
+  //. > S.when(x => x < 0, Math.abs, 4)
+  //. 4
+  //. ```
+  function when(pred, f, x) {
+    return ifElse(pred, f, I, x);
+  }
+  S.when = def('when', {}, [Pred(a), Fn(a, b), a, b], when);
+
+  //# unless :: (a -> Boolean) -> (a -> b) -> a -> b
+  //.
+  //. Takes a unary predicate, a unary function, and a value of any type, and
+  //. returns the result of applying the function to the value if the value
+  //. does not satisfy the predicate; the value otherwise.
+  //.
+  //. ```javascript
+  //. > S.unless(x => x < 0, Math.sqrt, -1)
+  //. -1
+  //.
+  //. > S.unless(x => x < 0, Math.sqrt, 16)
+  //. 4
+  //. ```
```

Comment:
I have a slight preference for showing the example that applies the `a -> b` function first; the failure case second. What do you think?